lib/http/uri_normalizer.rb

```diff
@@ -0,0 +1,25 @@
+require "uri"
+
+module HTTP
+  module URINormalizer
+    module_function
+
+    # Normalize URI
+    #
+    # @param [#to_s] uri
+    # @return [::URI]
+    def normalize_uri(uri)
```

Comment:
If we're going to support non-ASCII, we should fully support them imo. Eg, `http://🚚.la` doesn't work for URI, while it does for Addressable.